### PR TITLE
Use keytar to store connection string

### PR DESCRIPTION
**Pull request by formulahendry**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -725,6 +725,7 @@
   },
   "devDependencies": {
     "@types/fs-extra": "^5.0.4",
+    "@types/keytar": "^4.4.0",
```
Comment (adashen) [inline package.json]: only need dev dependencies?

File: src/credentialStore.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+"use strict";
+import * as keytarType from "keytar";
```
Comment (adashen) [inline src/credentialStore.ts]: keytar is not added as dependency right?

File: src/credentialStore.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+"use strict";
+import * as keytarType from "keytar";
+import * as vscode from "vscode";
+import { Constants } from "./constants";
+
+export class CredentialStore {
+    public static getPassword(account: string): Promise<string> {
+        try {
+            return this.keytar.getPassword(Constants.ExtensionId, account);
```
Comment (adashen) [inline src/credentialStore.ts]: If keytar could not get the value of input key, its null or throw error?

Comment (formulahendry) [reply src/credentialStore.ts]: It's null

File: package.json (hunk 1 above)
Comment (formulahendry) [reply package.json]: Yes, keytar is shipped with VS Code.

File: src/credentialStore.ts (hunk 2 above)
Comment (formulahendry) [reply src/credentialStore.ts]: Yes, keytar is shipped with VS Code.

File: src/credentialStore.ts (hunk 3 above)
Comment (adashen) [reply src/credentialStore.ts]: error is only thrown when keystar is null?

File: src/credentialStore.ts (hunk 2 above)
Comment (adashen) [reply src/credentialStore.ts]: From getCoreNodeModule, it seems keytar is not in normal nodemodule, So shall we import here? From sample in https://code.visualstudio.com/api/advanced-topics/remote-extensions, it doesn't import keytar

File: src/credentialStore.ts (hunk 3 above)
Comment (formulahendry) [reply src/credentialStore.ts]: error is only thrown when

- keystar is null
- OS does not have a credential store

Comment (adashen) [reply src/credentialStore.ts]: OS does not have a credential store will make keystar null right?

File: src/credentialStore.ts (hunk 2 above)
Comment (formulahendry) [reply src/credentialStore.ts]: Importing keytarType is just for type (for intellisense), refer to https://github.com/Microsoft/vscode-azure-ext-sdk#access-os-credential-stores

File: src/credentialStore.ts (hunk 3 above)
Comment (formulahendry) [reply src/credentialStore.ts]: I don't think so.  keystar is just a npm package. If there is no credential store, keystar could still exist. But when there is no credential store, this.keytar.getPassword() will throw error.

**Pull request merged**